docs/optimize/pipeline_frequency_optimizer.md

```diff
@@ -103,6 +103,36 @@ As another example, in Megatron-LM, users can pass in their custom `forward_step
 1. Wrap the forward pass region with `opt.on_instruction_begin("forward")` and `opt.on_instruction_end("forward")`.
 1. Wrap the backward pass region with `opt.on_instruction_begin("backward")` and `opt.on_instruction_end("backward")`.
 
+
+## Profiler Integration
+
+### InstructionProfiler
+
+**Description:**  
+The InstructionProfiler iterates over a range of frequencies to profile the time and energy consumption of forward and backward instructions in each pipeline stage. It aggregates the measurements and writes a CSV file with the average values for each instruction at each frequency. Use this profiler when you want to obtain detailed performance data for optimizing GPU frequency.
+
+### 3D-Parallel Support
+
+#### make_3d_parallel
+
+**Description:**  
+The make_3d_parallel wrapper enables 3D parallelism support by aggregating profiling results from ranks sharing the same pipeline-parallel rank (`pp_rank`) and replicating per-stage schedules across those ranks.
+
+**Wrappers:**  
+- PointSolution3D  
+- InstructionProfiler3D
+
+## Scheduler Instantiation
+
+Scheduler instantiation is fully managed by PFOServerSettings. The `scheduler` field (typed as a `PyObject`) in PFOServerSettings is automatically imported based on the environment variable `ZEUS_PFO_SCHEDULER` (or its default value). You can choose the scheduler by setting this environment variable to either `"InstructionProfiler"` for profiling or `"PointSolution"` for fixed scheduling. Additionally, any extra parameters specific to the scheduler can be provided through the `scheduler_args` field.
+
+For example, in your job manager you can instantiate the scheduler as follows:
+
+```python
+scheduler = pfo_settings.scheduler(job_info, rank_infos, pfo_settings, **pfo_settings.scheduler_args)
+```
+If you wish to perform profiling to gather detailed performance data, set `ZEUS_PFO_DUMP_DATA=true` and   `ZEUS_PFO_SCHEDULER=InstructionProfiler`. Otherwise, for fixed scheduling, set `ZEUS_PFO_SCHEDULER=PointSolution`.
+
```

Comment:
Delete the whole thing and replace with instructions on how to run the server, like the code block in https://ml.energy/zeus/optimize/pipeline_frequency_optimizer/#choose-and-start-training